### PR TITLE
Run3-alca217 Substitute the older AlCa code for Isotracks and HBHEMuon with the newer version

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Substitute the older AlCa code for Isotracks and HBHEMuon with the newer version requiring smaller disk space for the AlCaReco files

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
Thanks @bsunanda although this will not be enough in itself, we'll need to change the T0 config too. 
I assume this is not relevant for MWGR and CRAFT, only for collisions, so we'll have to come back to this when we are finalizing the AlCaRECO matrix for collisions.

**Comment by tvami** [comment]
Hi @bsunanda please add the new AlCaRECO either to a new relval wf or replace `HcalCalIsoTrkFilter` in
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L2537
and
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L2729
Thanks!

**Comment by tvami** [comment]
> I shall do that in a separate PR if that works for you

Hmm, I'd really prefer doing it here, otherwise the tests we are running are not actually running the changes you made, so we are testing nothing current :)

**Comment by tvami** [comment]
I think 138.4 is not part of the standard Jenkins tests, at least I dont see it here
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0db515/22314/runTheMatrix-results/

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from Configuration.AlCa.autoCond import autoCond
 process.GlobalTag.globaltag=autoCond['run2_data']
 
-process.load("RecoLocalCalo.EcalRecAlgos.EcalSeverityLevelESProducer_cfi")
+#process.load("RecoLocalCalo.EcalRecAlgos.EcalSeverityLevelESProducer_cfi")
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py:16]: Please delete this line if the tests work out like this.

Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc [hunk 2]
```diff
@@ -130,10 +130,13 @@ class AlCaHcalHBHEMuonProducer : public edm::stream::EDProducer<edm::GlobalCache
...
   edm::EDGetTokenT<reco::MuonCollection> tok_Muon_;
 
-  edm::ESGetToken<HcalDDDRecConstants, HcalRecNumberingRecord> tok_ddrec_;
+  edm::ESGetToken<HcalDDDRecConstants, HcalRecNumberingRecord> tok_ddrec0_;
+  edm::ESGetToken<HcalDDDRecConstants, HcalRecNumberingRecord> tok_ddrec1_;
+  edm::ESGetToken<HcalRespCorrs, HcalRespCorrsRcd> tok_respcorr0_;
+  edm::ESGetToken<HcalRespCorrs, HcalRespCorrsRcd> tok_respcorr1_;
+  edm::ESGetToken<CaloGeometry, CaloGeometryRecord> tok_geom0_;
+  edm::ESGetToken<CaloGeometry, CaloGeometryRecord> tok_geom1_;
   edm::ESGetToken<HcalTopology, HcalRecNumberingRecord> tok_htopo_;
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc:139]: all these could be `const`-s

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc [hunk 3]
```diff
@@ -190,10 +194,13 @@ AlCaHcalHBHEMuonProducer::AlCaHcalHBHEMuonProducer(const edm::ParameterSet& iCon
...
+  tok_ddrec0_ = esConsumes<HcalDDDRecConstants, HcalRecNumberingRecord, edm::Transition::BeginRun>();
+  tok_ddrec1_ = esConsumes<HcalDDDRecConstants, HcalRecNumberingRecord>();
+  tok_respcorr0_ = esConsumes<HcalRespCorrs, HcalRespCorrsRcd, edm::Transition::BeginRun>();
+  tok_respcorr1_ = esConsumes<HcalRespCorrs, HcalRespCorrsRcd>();
+  tok_geom0_ = esConsumes<CaloGeometry, CaloGeometryRecord, edm::Transition::BeginRun>();
+  tok_geom1_ = esConsumes<CaloGeometry, CaloGeometryRecord>();
+  tok_htopo_ = esConsumes<HcalTopology, HcalRecNumberingRecord>();
   tok_magField_ = esConsumes<MagneticField, IdealMagneticFieldRecord>();
   tok_chan_ = esConsumes<EcalChannelStatus, EcalChannelStatusRcd>();
   tok_sevlv_ = esConsumes<EcalSeverityLevelAlgo, EcalSeverityLevelAlgoRcd>();
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc:206]: and all these should be moved to the constructor here
https://github.com/cms-sw/cmssw/blob/f7f3c5a38e71e6835a7bf89aa531aa694a629d6f/Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc#L157

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@bsunanda before we abuse Jenkins further with the testing, can you please make sure the tests in https://github.com/cms-sw/cmssw/pull/36922#issuecomment-1037399882
for example 11634.0 pass your local test?

**Comment by bsunanda** [comment]
It always pass my local tests. Also tried 11634.0 and it passed all 5 steps
________________________________

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@tvami I have squashed the commits and outputs are ok

**Comment by tvami** [comment]
+alca
  * tests look good, the new alcarecos can be found in the relval outputs

**Comment by bsunanda** [comment]
@perrotta,@qliphy, @jordan-martins, @bbilin, @wajidalikhan Please sign and merge this PR

**Comment by srimanob** [comment]
+Upgrade

There is an update on sequence defined in `Configuration/PyReleaseValidation` which inline with PR description.

**Comment by tvami** [comment]
urgent
 * as @bsunanda mentioned just now this is needed for pre5

**Comment by bbilin** [comment]
+pdmv

File: Configuration/AlCa/python/autoAlca.py [hunk 4]
```diff
@@ -34,7 +34,7 @@
                   "DoubleMu" : "MuAlCalIsolatedMu+MuAlOverlaps+TkAlZMuMu",
                   "DoubleMuParked" : "MuAlCalIsolatedMu+MuAlOverlaps+TkAlZMuMu",
                   "MuOniaParked" : "TkAlJpsiMuMu+TkAlUpsilonMuMu",
-                  "DoubleElectron" : "EcalCalZElectron+EcalUncalZElectron+HcalCalIsoTrkFilter",
+                  "DoubleElectron" : "EcalCalZElectron+EcalUncalZElectron",
```
Comment (perrotta) [inline Configuration/AlCa/python/autoAlca.py:37]: No replacement here, simply removed?

Comment (tvami) [reply Configuration/AlCa/python/autoAlca.py:37]: @bsunanda

Comment (bsunanda) [reply Configuration/AlCa/python/autoAlca.py:37]: Yes. We do not need this AlCaReco stream for IsoTrack. We only need this from JetHT and DoubleEgamma

**Comment by perrotta** [comment]
+1

**Pull request merged**